code/modules/mob/living/carbon/xenomorph/castes/baneling/abilities_baneling.dm

```diff
@@ -0,0 +1,127 @@
+// ***************************************
+// *********** Baneling Explode
+// ***************************************
+/datum/action/xeno_action/baneling_explode
+	name = "Baneling Explode"
+	action_icon_state = "baneling_explode"
+	desc = ""
+	ability_name = "baneling explode"
+	keybinding_signals = list(
+		KEYBINDING_NORMAL = COMSIG_XENOABILITY_BANELING_EXPLODE,
+	)
+
+/datum/action/xeno_action/baneling_explode/give_action(mob/living/L)
+	. = ..()
+	var/mob/living/carbon/xenomorph/baneling/X = L
```

Comment:
This should be just xenomorph, not baneling specific. Same for the rest of the file.